system/src/system_connection_manager.h

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef SYSTEM_CONNECTION_MANAGER_H
+#define SYSTEM_CONNECTION_MANAGER_H
+
+#include "hal_platform.h"
+
+#if HAL_PLATFORM_IFAPI
+
+#include "system_network.h"
+#include "spark_wiring_vector.h"
+
+namespace particle { namespace system {
+
+struct ConnectionMetrics {
```

Comment:
Nit/TODO: this should probably be a proper C++ class with constructor, reset method etc.